sdk/iot/hub/tests/cmocka/test_az_iot_hub_client.c

```diff
@@ -554,6 +591,141 @@ static void test_az_iot_hub_client_properties_append_twice_small_buffer_fail(voi
   assert_int_equal(az_span_capacity(props._internal.properties), sizeof(test_correct_two_key_value) - 2);
 }
 
+static void test_az_iot_hub_client_properties_find_succeed(void** state)
+{
+  (void)state;
+
+  az_span test_span = az_span_from_str(TEST_KEY_VALUE_ONE);
+  az_iot_hub_client_properties props;
+
+  assert_int_equal(az_iot_hub_client_properties_init(&props, test_span), AZ_OK);
+
+  az_span out_value;
+  assert_int_equal(az_iot_hub_client_properties_find(&props, test_key_one, &out_value), AZ_OK);
+  assert_memory_equal(
+      az_span_ptr(out_value), az_span_ptr(test_value_one), (size_t)az_span_length(test_value_one));
+}
+
+static void test_az_iot_hub_client_properties_find_middle_succeed(void** state)
+{
+  (void)state;
+
+  az_span test_span = az_span_from_str(TEST_KEY_VALUE_THREE);
+  az_iot_hub_client_properties props;
+
+  assert_int_equal(az_iot_hub_client_properties_init(&props, test_span), AZ_OK);
+
+  az_span out_value;
+  assert_int_equal(az_iot_hub_client_properties_find(&props, test_key_two, &out_value), AZ_OK);
+  assert_memory_equal(
+      az_span_ptr(out_value), az_span_ptr(test_value_two), (size_t)az_span_length(test_value_two));
+}
+
+static void test_az_iot_hub_client_properties_find_end_succeed(void** state)
+{
+  (void)state;
+
+  az_span test_span = az_span_from_str(TEST_KEY_VALUE_TWO);
+  az_iot_hub_client_properties props;
+
+  assert_int_equal(az_iot_hub_client_properties_init(&props, test_span), AZ_OK);
+
+  az_span out_value;
+  assert_int_equal(az_iot_hub_client_properties_find(&props, test_key_two, &out_value), AZ_OK);
+  assert_memory_equal(
+      az_span_ptr(out_value), az_span_ptr(test_value_two), (size_t)az_span_length(test_value_two));
+}
+
+static void test_az_iot_hub_client_properties_find_substring_succeed(void** state)
+{
+  (void)state;
+
+  az_span test_span = az_span_from_str(TEST_KEY_VALUE_SUBSTRING);
```

Comment:
should also test "one_key=value_one&key=value_two"

Reply:
also what about "key_one=value_one&key_one=value_two"?

Reply:
Your first point is valid and I will add a test for that (and fix if it's a problem). As for the second point, is that something that this function should account for? If the user wanted they could use the `az_iot_hub_client_properties_next()` to iterate through the same named keys.

Reply:
updated and reworked using only `az_span_token()`

Reply:
>> If the user wanted they could use the az_iot_hub_client_properties_next() to iterate through the same named keys.

That is what I expect. Test case for it?

Reply:
In this case what would the test case be? That there are two names that are identical and this always returns the first one?

Reply:
humm, I figured that current_property was updated after the find

Reply:
No i think that should only be updated when using next. I think it might be kind of odd to search for a key, find it, underneath the current prop is update, and then another find would return no value. In the case where maybe someone calls it twice expecting the same value.